app/controllers/assessments_controller.rb

```diff
@@ -0,0 +1,69 @@
+class AssessmentsController < ApplicationController
+  before_action :set_contact, only: %i[new create]
+
+  def new
+    @users = User.all
+    @roles = Role.all
+    @type = %w[log schedule].include?(type_param) ? type_param : 'log'
+
+    @need = Need.new
+    @note = Note.new
+
+    @need.status = 'complete' if @type == 'log'
+    @need.user = current_user if @type == 'log'
+  end
+
+  def create
+    @type = %w[log schedule].include?(type_param) ? type_param : 'log'
+    if @type == 'log'
+      log_assessment
+    else
+      schedule_assessment
+    end
+  end
+
+  private
+
+  def set_contact
+    @contact = Contact.find(params[:contact_id])
+  end
+
+  def log_assessment
+    @need = Need.new(assessment_params.merge(contact_id: @contact.id, name: 'Logged Assessment'))
+    @note = Note.new(notes_params.merge(need: @need))
+    if @need.valid? && @note.valid? && @need.save && @note.save
+      redirect_to contact_path(@contact)
+      return
+    end
+
+    @users = User.all
+    @roles = Role.all
+    render :new
+  end
+
+  def schedule_assessment
+    @need = Need.new(assessment_params.merge(contact_id: @contact.id, name: 'Scheduled Assessment'))
```

Comment:
Hm, I think when scheduling an assessment, we want the user to be able to provide a description (e.g. "This person needs a check-in to see if they still need food delivered")